### DISCUSSION TITLE
Intel runtime library

### DISCUSSION THREAD

**Discussion by Peku995** [cdslaborg/paramonte#23] [Q&A]
Hi, I tried to install the runtime libray automatically via paramonte in Matlab, however, it does not work because somehow it seems as if it would use a wrong path. I also tried to install the library via downloading and installation, which worked, but paramonte does not detect the installation. 
What can I do to overcome this problem?

**Comment by fagheri** [comment]
Hi @Peku995 Which platform do you use? macOS? Linux? You should not need the Intel runtime library if you do not use the parallel MPI version. You may have to rebuild the library for Mac M1 and M2 architectures, but the [binaries in the latest release](https://github.com/cdslaborg/paramonte/releases) should work just fine with macOS amd64 architecture. Please update so I can escalate this discussion into an issue for resolution.

**Reply by Peku995** [reply]
I am using Windows. Somehow Paramonte downloaded the .exe file to install the library but could not install it. However, the command window showed the path to the downloaded file, so I could install it manually. 
However, the path was relatively well hidden among all the other output that Paramonte does. Perhaps it would be better if the path was displayed more prominently so that the user could see it more quickly.

**Reply by fagheri** [reply]
Thank you for clarifying. That seems odd because the ParaMonte library never generates a `.exe` file. Are you trying to build the ParaMonte::MATLAB library from the source files?
The MATLAB version should work fine, particularly on Windows. 
Can you unzip [the latest release](https://github.com/cdslaborg/paramonte/releases/download/v1.5.1/libparamonte_matlab_windows_x64.zip) and run the `main.m` example provided within a MATLAB session?
When you encounter the question *Do you wish to download and install the MPI runtime library  (only needed for parallel simulations) on your system now (y/n)?* type `n` and press enter. This will avoid any new file downloads from the web.

**Reply by fagheri** [reply]
Okay, I now see a potential bug in the library. When I type "y" in answer to the prompt question, I get the following error:
```

ParaMonte - WARNING: Exhausted all releases of the ParaMonte library in search 
ParaMonte - WARNING: of the prerequisites, but could not find: impi.dll
ParaMonte - WARNING: Please report this issue at 
ParaMonte - WARNING: 
ParaMonte - WARNING:     https://github.com/cdslaborg/paramonte/issues
ParaMonte - WARNING: 
ParaMonte - WARNING: In the meantime, visit, 
ParaMonte - WARNING: 
ParaMonte - WARNING:     https://www.cdslab.org/paramonte
ParaMonte - WARNING: 
ParaMonte - WARNING: for instructions to manually install the MPI library on your 
ParaMonte - WARNING: system. Aborting the automatic MPI installation by ParaMonte...

Unrecognized field name "fullFilePath".
Error in paramonte/installMPI (line 1890)
                                + "file location: " + self.prereqs.mpi.intel.fullFilePath;
Error in paramonte/verify (line 730)
                                succeeded = self.installMPI(); 
```

Is this the same problem you have encountered?
I will escalate this as an issue to fix.

**Reply by Peku995** [reply]
Thanks for the quick reply! I didn't mean that paramonte created an .exe file, but that it downloaded the library's .exe file. But it could not install it successfully. However, it did show me the link to the downloaded file in the output window. I then installed this file.

My output message looked different: It only said that it had successfully installed the library (which it hadn't), but at the same time the windows cmd window came up and said that the path I was looking for was not available.

**Reply by fagheri** [reply]
Now I have a much better understanding of the issue. I think the file name you mentioned is `w_mpi-rt_p_2019.4.245.exe`, which installs the Intel MPI runtime library. Based on this discussion, I have created a [new issue](https://github.com/cdslaborg/paramonte/issues/25#issue-1770863731) to resolve this potential bug. 
You can safely ignore the Intel MPI runtime library if you do not intend to run parallel simulations using MPI.
I believe @JoshuaOsborneDATA is working on a native shared-memory parallel version of this library for MATLAB that does not require MPI at all and uses MATLAB's parallel toolbox instead. This will further simplify parallel simulations.
This will appear in this library's impending new major release in the next few days.
Thank you again for reporting this potential bug; we greatly appreciate it!

**Reply by Peku995** [reply]
Using the matlab parallel toolbox sounds great!

**Comment by fagheri** [comment]
I just tested the library in MATLAB 2023, and it successfully finished the simulation. Here is an example output graph from the simulation I just did by running `main.m`.
![norm4d](https://github.com/cdslaborg/paramonte/assets/70309036/e74cb575-672c-4bf6-98c9-6aa5d90466b8)

**Comment by arwelHughes** [comment]
Yes!
________________________________
